src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/LegendItem.tsx

```diff
@@ -10,26 +11,51 @@ import { type ReputationChartDataItem } from '../types.ts';
 
 const displayName = 'v5.frame.ColonyHome.ReputationChart.partials.LegendItem';
 interface LegendItemProps {
-  chartItem:
-    | ReputationChartDataItem
-    | {
-        color: string;
-        label: string;
-        value: undefined;
-        shouldTruncateLegendLabel?: boolean;
-      };
+  chartItem: ReputationChartDataItem;
 }
 
 const LEGEND_LABEL_LENGTH = 12;
 
+const LegendItemOuter: FC<
+  PropsWithChildren<{ id?: string; searchParam?: string }>
+> = ({ children, id, searchParam }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const handleClick = useCallback(() => {
+    if (id && searchParam) {
+      searchParams.set(searchParam, id);
+      setSearchParams(searchParams);
+    }
+  }, [id, searchParam, searchParams, setSearchParams]);
+  if (searchParam) {
+    return (
+      <button
+        type="button"
+        className="flex flex-row items-center gap-1"
+        onClick={handleClick}
+      >
+        {children}
+      </button>
+    );
+  }
+  return <div className="flex flex-row items-center gap-1">{children}</div>;
+};
+
```

Comment:
I would suggest moving this component to a separate file

Reply:
Thank you, I'll do that. I was lazy :see_no_evil: